### ISSUE TITLE
Option to automatically switch to upper case on a new sentence

### ISSUE THREAD

**Issue by CaptainBlagbird** [8VIM/8VIM#21]
The app would be even a little bit more efficient if it started in upper case mode and the mode would switch to uppercase automatically after a period and space was just written.

This behaviour should preferably be enabled/disabled in a settings menu.

**Comment by flide** [comment]
Good point. Noted.

**Comment by WuerfelDev** [comment]
Also _Double-space full stop_: Two spaces become '. '

**Comment by flide** [comment]
I think we should not hard code double spaces to replace to a '. ' 
But I think that could be a good default for "Abbreviation Expansion" feature etc.

**Comment by Glitchy-Tozier** [comment]
Yeah, add options like the double-space one, but please keep it optional.

**Comment by doak** [comment]
> Also Double-space full stop: Two spaces become '. '

I like the possibility to write with this keyboard "literally", i.e. without a dictionary getting in my way. If I want to write two spaces, then please insert them. It even vote against an option (if I could vote), though it would not harm in the first place. But cluttering the preferences with very specific use cases is not useful at the end.
The idea of an configurable abbreviation/replacement list (#74) is a generic approach to this, can be adapted for many other use cases as well and is for that reason imho a great approach.

The same is true for automatic capitalisation after full stop. I always switch it off (on the regular keyboard) because it gets in may way and is wrong the most time anyway. What happens after abbreviations like `etc.` etc.?
This is for sure biased by my language, which uses capitalised words for nouns as well. (which is great btw :))

I agree that there is a need to enter punctuation more quickly, but don't confuse heuristic guessing with speed.

**Comment by lxne** [comment]
Maybe it would be an option to make two spaces into an '. ' but three spaces into two spaces, four spaces into three spaces and so on. So if you want more spaces you would have do do always one more touch than you want spaces. But you can make a '. ' with two spaces. You'd be very fast for . and for spaces. I am against guessing, too.

**Comment by martin-braun** [comment]
#330 could make this issue irrelevant for some people, also @WuerfelDev / @lxne #265 might be interesting for you.

**Comment by ecarl65** [comment]
I'm new to this keyboard but the main thing I'm missing is this auto-capitalization at the beginning of a sentence.
